packages-private/dts-test/defineComponent.test-d.tsx

```diff
@@ -118,7 +118,7 @@ describe('with object props', () => {
     // default + function
     ffff: {
       type: Function as PropType<(a: number, b: string) => { a: boolean }>,
-      default: (a: number, b: string) => ({ a: a > +b }),
+      default: () => (a: number, b: string) => ({ a: a > +b }),
```

Comment:
Please avoid changing the existing test cases, as it breaks the current behavior. Let's focus on fixing the issue without introducing a breaking change.

Reply:
I think it's a bug.

> Object or array defaults must be returned from a factory function. The function receives the raw props received by the component as the argument.

Function is also object.

Reply:
Function-type props are handled differently and don't require a factory function. You can find examples in the docs: https://vuejs.org/guide/components/props.html#prop-validation

```ts
// Function with a default value
propH: {
  type: Function,
  // Unlike object or array defaults, this is not a factory
  // function - this is a function serving as the default value
  default() {
    return 'Default function'
  }
}
```

Reply:
This brings great difficulty to type inference.